### ISSUE TITLE
Unsafe Handling of DateFormatter

### ISSUE THREAD

**Issue by sidepelican** [hummingbird-project/hummingbird#273]
## Problem

The `ISO8601DateFormatter` in `swift-corelibs-foundation` is not thread-safe.
However, `URLEncodedFormDecoder` and `URLEncodedFormEncoder` use this class as a singleton. This can cause a runtime crash, such as a segmentation fault.

Reference to the code:
https://github.com/hummingbird-project/hummingbird/blob/d7fb66e1a8d6bc815d335e794910b1d43c8b3e07/Sources/HummingbirdFoundation/Codable/URLEncodedForm/URLEncodedForm.swift#L51-L55

## Steps to Reproduce

The following code, when run multiple times in a Linux Swift environment, sometimes causes a crash:

```swift
import Foundation

let iso8601Formatter: ISO8601DateFormatter = {
  let formatter = ISO8601DateFormatter()
  formatter.formatOptions = .withInternetDateTime
  return formatter
}()

let now = Date()
for i in 0..<1000 {
  DispatchQueue.global().async {
    let date = now.addingTimeInterval(Double(i) * 60 * 60 * 24)
    print(iso8601Formatter.string(from: date))
  }
}

sleep(1)
```

## Tested Environment

- Swift 5.9 (swift-5.9-RELEASE)
- Target: x86_64-unknown-linux-gnu

**Comment by adam-fowler** [comment]
I couldn't get it to crash, but I've still got rid of the global instance.
Fixed in #274
I thought this had been fixed in corelibs foundation, but looking through issues it doesn't look like it has. I doubt it will now and we'll have to wait for the new Foundation

**Comment by adam-fowler** [comment]
Fix in v1.9.1
